src/wavestate/control/MIMO/util.py

```diff
@@ -39,6 +40,8 @@ def io_normalize(io_arg, Nmax):
         if isinstance(io_arg, (list, tuple)):
             # convert to a dictionary
             io_arg = {k: i for i, k in enumerate(io_arg)}
+        elif isinstance(io_arg, dict):
```

Comment:
This should be tested against a collections.Mapping rather than a dict. Otherwise it misses Bunches and similar